src/tests/CompanyDetail.test.js

```diff
@@ -0,0 +1,38 @@
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { companiesData } from '../redux/company/CompaniesSlice';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+
+describe('companiesSlice actions and reducers', () => {
+  it('should dispatch companiesData.fulfilled action with the correct payload', async () => {
+    const initialState = {
+      companies: [],
+      companyDetails: [],
+      isLoading: false,
+      error: false,
+    };
+
+    const store = mockStore(initialState);
+
+    await store.dispatch(companiesData());
+
+    expect(store.getActions()).toMatchSnapshot();
+  });
+
+  it('should dispatch companiesData.rejected action with the correct error message', async () => {
+    const initialState = {
+      companies: [],
+      companyDetails: [],
+      isLoading: false,
+      error: false,
+    };
+
+    const store = mockStore(initialState);
+
+    await store.dispatch(companiesData());
+
+    expect(store.getActions()).toMatchSnapshot();
+  });
```

Comment:
- [ ] These 2 tests are currently failing. Please try to make them green ✔️ 
![image](https://github.com/Surafels/Metric-webapp/assets/46839548/b4e802e3-bee1-490c-b97f-22b6ea1fe7e9)